test/common/stats/stat_test_utility.cc

```diff
@@ -123,6 +100,39 @@ void forEachSampleStat(int num_clusters, std::function<void(absl::string_view)>
   }
 }
 
+MemoryTest::Mode MemoryTest::mode() {
+#if !defined(TCMALLOC) || defined(ENVOY_MEMORY_DEBUG_ENABLED)
+  // We can only test absolute memory usage if the malloc library is a known
+  // quantity. This decision is centralized here. As the preferred malloc
+  // library for Envoy is TCMALLOC that's what we test for here. If we switch
+  // to a different malloc library than we'd have to re-evaluate all the
+  // thresholds in the tests referencing MemoryTest.
+  return Mode::Disabled;
+#else
+  // Even when using TCMALLOC is defined, it appears that
+  // Memory::Stats::totalCurrentlyAllocated() does not work as expected
+  // on some platforms, so try to force-allocate some heap memory
+  // and determine whether we can measure it.
+  const size_t start_mem = Memory::Stats::totalCurrentlyAllocated();
+  std::string long_string("more than 22 chars to exceed libc++ short-string optimization");
```

Comment:
Will this be optimized out by compiler? Would adding `volatile` help?

Reply:
I do have the assert below which proves that it isn't currently, but I added the 'volatile' keyword in case that helps avoid bumping into that with an awesome future compiler.